pkg/virt-handler/vm.go

```diff
@@ -1205,7 +1243,10 @@ func (d *VirtualMachineController) processVmUpdate(origVMI *v1.VirtualMachineIns
 		}
 		d.recorder.Event(vmi, k8sv1.EventTypeNormal, v1.PreparingTarget.String(), "VirtualMachineInstance Migration Target Prepared.")
 	} else if d.isMigrationSource(vmi) {
-		err = client.MigrateVirtualMachine(vmi)
+		if !isVmDisksMigratable {
+			return fmt.Errorf("unable to migrate the vm, mixes shared and non-shared volumes.")
+		}
+		err = client.MigrateVirtualMachine(vmi, isBlockMigrationRequired)
```

Comment:
I would have expected there a more general context, which e.g. passes through all `PVCs`. Would this approach still be able to cover a mixture of differnt disks? block and file system based?

Reply:
I was debating with myself about this... I didn't find a good enough reason why to pass the `PVCs` instead of a boolean. Seems like a boolean should be enough to indicate if its a block migration or not.
Also, another issue (probably solvable) was that we are replacing filesystem `PVCs` with a `HostDisk` which may make things less straightforward if we will pass `PVCs` to virt-launcher.
What do you think?

Reply:
> Also, another issue (probably solvable) was that we are replacing filesystem `PVCs` with a `HostDisk` which may make things less straightforward if we will pass `PVCs` to virt-launcher.
> What do you think?

Yeah. If we pass them through anyway, we can change it. 

> I was debating with myself about this... I didn't find a good enough reason why to pass the `PVCs` instead of a boolean. Seems like a boolean should be enough to indicate if its a block migration or not.

Here I simply may not understand the context. What if one PVC is file-based and one is block based? What if another disk is read-only and ephemeral (e.g. registry disk)? What exactly is this boolean indicating then?